caddyhttp/proxy/body.go

```diff
@@ -23,9 +31,19 @@ func (b *bufferedBody) rewind() error {
 	return err
 }
 
+type unbufferedBody struct {
+	io.ReadCloser
+}
+
+var _ rewindableReader = &unbufferedBody{}
+
+func (b *unbufferedBody) rewind() error {
+	return errors.New("cannot rewind unbuffered body")
```

Comment:
Oh, you're suggesting to just make this a no-op?

If it's a no-op, maybe the bigger design question is, Why does `unbufferedBody` implement `rewindableReader` if it is not a rewindable reader?